### ISSUE TITLE
Trackers removed after restart.

### ISSUE THREAD

**Issue by hpeyerl** [Jackett/Jackett#1968]
I've been finding about once a week, my trackers disappear. Once I re-add them, everything works fine.   Platform is Ubuntu 16.04 with Mono 5.0.1.1.   It would appear the problem is certificate related but I've tried removing the certificates and reinstalling them.  I've also tried linking $HOME/.config/.mono to /etc/mono/certstore which is what Sonarr is happily using. 

Since the trackers work fine if I add them manually after Jackett gets restarted (usually I notice a week later when shows are not found), I'm at a loss for knowing where to start looking for the problem.

Here's the startup output.  You can see Jackett deleting trackers:

10-15 12:05:45 Info Starting Jackett 0.8.259.0
10-15 12:05:45 Info Environment version: 4.0.30319.42000 (/usr/lib/mono/4.5)
10-15 12:05:45 Info OS version: Unix 4.4.35.1 (64bit OS) (64bit process)
10-15 12:05:45 Info ThreadPool MaxThreads: 200 workerThreads, 200 completionPortThreads
10-15 12:05:45 Info issue: Ubuntu 16.04 LTS \n \l
10-15 12:05:45 Info mono version: 5.0.1.1 (2017-02/5077205 Thu May 25 09:19:18 UTC 2017)
10-15 12:05:45 Info Using HTTP Client: HttpWebClient
10-15 12:05:45 Error Failed loading configuration for TorrentDay, trying backup System.Security.Cryptography.CryptographicException: Data unprotection failed. ---> System.Security.Cryptography.CryptographicException: Invalid data.
  at Mono.Security.Cryptography.ManagedProtection.Unprotect (System.Byte[] encryptedData, System.Byte[] optionalEntropy, System.Security.Cryptography.DataProtectionScope scope) [0x00241] in <b03484857fd24880bc008651240c8364>:0
  at System.Security.Cryptography.ProtectedData.Unprotect (System.Byte[] encryptedData, System.Byte[] optionalEntropy, System.Security.Cryptography.DataProtectionScope scope) [0x00023] in <b03484857fd24880bc008651240c8364>:0
   --- End of inner exception stack trace ---
  at System.Security.Cryptography.ProtectedData.Unprotect (System.Byte[] encryptedData, System.Byte[] optionalEntropy, System.Security.Cryptography.DataProtectionScope scope) [0x0003f] in <b03484857fd24880bc008651240c8364>:0
  at Jackett.Services.ProtectionService.UnProtectDefaultMethod (System.String plainText) [0x00121] in <431b593737d5408c91cdc3cdb49f985b>:0
  at Jackett.Services.ProtectionService.UnProtect (System.String plainText) [0x00011] in <431b593737d5408c91cdc3cdb49f985b>:0
  at Jackett.Models.IndexerConfig.ConfigurationData.LoadValuesFromJson (Newtonsoft.Json.Linq.JToken json, Jackett.Services.IProtectionService ps) [0x0011c] in <431b593737d5408c91cdc3cdb49f985b>:0
  at Jackett.Indexers.BaseIndexer.LoadValuesFromJson (Newtonsoft.Json.Linq.JToken jsonConfig, System.Boolean useProtectionService) [0x0000f] in <431b593737d5408c91cdc3cdb49f985b>:0
  at Jackett.Indexers.BaseIndexer.LoadFromSavedConfiguration (Newtonsoft.Json.Linq.JToken jsonConfig) [0x0000b] in <431b593737d5408c91cdc3cdb49f985b>:0
  at Jackett.Services.IndexerConfigurationService.Load (Jackett.Indexers.IIndexer idx) [0x00021] in <431b593737d5408c91cdc3cdb49f985b>:0
10-15 12:05:45 Error Failed loading backup configuration for TorrentDay, you must reconfigure this indexer System.Security.Cryptography.CryptographicException: Data unprotection failed. ---> System.Security.Cryptography.CryptographicException: Invalid data.
  at Mono.Security.Cryptography.ManagedProtection.Unprotect (System.Byte[] encryptedData, System.Byte[] optionalEntropy, System.Security.Cryptography.DataProtectionScope scope) [0x00241] in <b03484857fd24880bc008651240c8364>:0
  at System.Security.Cryptography.ProtectedData.Unprotect (System.Byte[] encryptedData, System.Byte[] optionalEntropy, System.Security.Cryptography.DataProtectionScope scope) [0x00023] in <b03484857fd24880bc008651240c8364>:0
   --- End of inner exception stack trace ---
  at System.Security.Cryptography.ProtectedData.Unprotect (System.Byte[] encryptedData, System.Byte[] optionalEntropy, System.Security.Cryptography.DataProtectionScope scope) [0x0003f] in <b03484857fd24880bc008651240c8364>:0
  at Jackett.Services.ProtectionService.UnProtectDefaultMethod (System.String plainText) [0x00121] in <431b593737d5408c91cdc3cdb49f985b>:0
  at Jackett.Services.ProtectionService.UnProtect (System.String plainText) [0x00011] in <431b593737d5408c91cdc3cdb49f985b>:0
  at Jackett.Models.IndexerConfig.ConfigurationData.LoadValuesFromJson (Newtonsoft.Json.Linq.JToken json, Jackett.Services.IProtectionService ps) [0x0011c] in <431b593737d5408c91cdc3cdb49f985b>:0
  at Jackett.Indexers.BaseIndexer.LoadValuesFromJson (Newtonsoft.Json.Linq.JToken jsonConfig, System.Boolean useProtectionService) [0x0000f] in <431b593737d5408c91cdc3cdb49f985b>:0
  at Jackett.Indexers.BaseIndexer.LoadFromSavedConfiguration (Newtonsoft.Json.Linq.JToken jsonConfig) [0x0000b] in <431b593737d5408c91cdc3cdb49f985b>:0
  at Jackett.Services.IndexerConfigurationService.Load (Jackett.Indexers.IIndexer idx) [0x00079] in <431b593737d5408c91cdc3cdb49f985b>:0

**Comment by kaso17** [comment]
I suspect jackett can't write to $HOME/.config/.mono

**Comment by hpeyerl** [comment]
I had previously confirmed that $HOME/.config/.mono were owned by me (same user as mono is running as) and perms were 700 on the directory and 600 on the files.   At some point, however, I believe I deleted it and reinstalled but ran it as root.  So when I just checked, it was owned by root.  Just 'chown -R'd it to me and it seems to be working even after a couple of reboots.  However, I've also deleted and reinstalled Jackett a few times and at least one of those I ran it as root.  Prior to my debugging this issue, it was definitely owned and writable as me but I was having this issue.  I'll close this issue and monitor.

Thanks.
